### PR TITLE
Feature/sms states streams

### PR DESCRIPTION
**Pull request by joanpablo**
Hi @babariviere , I would like you to check this implementation of 'sms delivery'. Your implementation was very very good, I just added some minnor new concepts as:

1-) Using Dart Streams and EventChannels for sms delivered so you can subscribe multiple listeners to a specific sms status change

2-) I added a Dtream to the SmsSender class (onSmsDelivered) so you can be notified for any delivered message regardless of the message (in the example app code I used it to be notified of sms delivered from other view rather than send).

3-) Some Android channel implementation changes

### CODE REVIEW

File: lib/contact.dart [hunk 1]
```diff
@@ -188,8 +187,9 @@ class UserProfile {
           new Photo(Uri.parse(data["photo"]), Uri.parse(data["thumbnail"]));
     }
     if (data.containsKey("addresses")) {
-      _addresses = new List<String>();
-      _addresses.addAll(data["addresses"]);
+      for (final address in data["addresses"]) {
+        _addresses.add(address);
+      }
```
Comment (babariviere) [inline lib/contact.dart]: What is the difference between addAll and this iter?

Comment (joanpablo) [reply lib/contact.dart]: Well, it throw some exception related to casting problems, List<dynamic> is not subtype of Iterable<String>

Comment (joanpablo) [reply lib/contact.dart]: whe the list is 'List<dynamic>' when expecting 'Iterable<String>'

Comment (babariviere) [reply lib/contact.dart]: Ok thanks

Comment (joanpablo) [reply lib/contact.dart]: List of type dynamic and Iterable of type String

Comment (joanpablo) [reply lib/contact.dart]: I will resolve conflicts this night, have a good day.

**Comment by babariviere** [comment]
:ok_hand:

**Comment by joanpablo** [comment]
Hahaha thanks, how are you doing? Tell me about your studies, did you
already finished the University? Did you made the startup?

**Comment by joanpablo** [comment]
Hi Bastièn, what do you think about upgrade to 0.1.1 and upload to pub
repo, dor increasing activity?

**Comment by babariviere** [comment]
Yes you are right!

**Comment by babariviere** [comment]
Can you try to push the upgrade please? That will make you a contributor on flutter to

**Comment by babariviere** [comment]
I have added you as an uploader on pub.dartlang.org

**Comment by joanpablo** [comment]
Yes of course, thanks right now I'm in a meeting, I very glad to do it.
Thanks again

**Pull request merged**